File: examples/middleware/modify_response/modify_response.go
package main

import (
	"bufio"
	"encoding/json"
	"log"
	"os"

	"github.com/SpectoLabs/hoverfly/core/handlers/v2"
)

func main() {
	l := log.New(os.Stderr, "", 0)

	s := bufio.NewScanner(os.Stdin)
	for s.Scan() {

		var payload v2.RequestResponsePairViewV1

		err := json.Unmarshal(s.Bytes(), &payload)
		if err != nil {
			l.Println("Failed to unmarshal payload from hoverfly")
		}

		payload.Response.Body = "body was replaced by middleware\n"

		bts, err := json.Marshal(payload)
		if err != nil {
			l.Println("Failed to marshal new payload")
		}

		os.Stdout.Write(bts)
	}
}

File: examples/middleware/modify_response/modify_response.rb
#!/usr/bin/env ruby
# encoding: utf-8

require 'rubygems'
require 'json'

while payload = STDIN.gets
  next unless payload

  jsonPayload = JSON.parse(payload)

  jsonPayload["response"]["body"] = "body was replaced by middleware\n"

  STDOUT.puts jsonPayload.to_json

end